### ISSUE TITLE
Tilde in Redirect URL Gets Removed After Redirect

### ISSUE THREAD

**Issue by ninjawarrior1337** [clerkinc/javascript#1253]
- [x] Review the documentation: https://clerk.com/docs
- [x] Search for existing issues: https://github.com/clerkinc/javascript/issues
- [x] Go through package changelog files.
- [x] Provide the Frontend API key from your application dashboard.
`pk_test_Z2xvd2luZy1tb2xseS03MC5jbGVyay5hY2NvdW50cy5kZXYk
`

## Package + Version

- [x] `@clerk/clerk-react@4.17.0`

### Dependencies + versions

```json
{
"dependencies": {
    "@clerk/clerk-react": "^4.17.0",
    "@types/react": "18.2.7",
    "@types/react-dom": "18.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0"
  },
  "devDependencies": {
    "react-scripts": "latest"
  }
}
```

### Browser/OS

If applicable e.g. `Chrome {version}`, `Node {version}` , `Bun {version}` , `Cloudflare worker {version}`
`Chrome 113.0.5672.126 (Official Build) (64-bit)`
`Node v18.12.1`

## Description
When a tilde is present in the redirectUrl prop using the sign-in component (and other components that will redirect to a specified path after auth), the given path is not respected and will redirect to a URL with the tilde removed.

Ex: (Before Redirect) https://example.com/~test123/main -> https://example.com/test123/main (After Redirect)

Example Sandbox: https://stackblitz.com/edit/stackblitz-starters-tp8iww?file=src%2FApp.tsx

**Comment by dimkl** [comment]
Hello @ninjawarrior1337.
Thank you for reporting this issue. 
Could you provide more information about how did you find this issue (eg is the `~` used as a convention for a framework or library)?

**Comment by ninjawarrior1337** [comment]
> Hello @ninjawarrior1337.
> Thank you for reporting this issue. 
> Could you provide more information about how did you find this issue (eg is the `~` used as a convention for a framework or library)?
> 

I'm a student at California State University Northridge. At the university, they allow students to host a public static site under the URL https://csun.edu/~{student_id}. Hence the requirement to have a ~ in the URL.

**Comment by clerk-cookie** [comment]
This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 10 days.

**Comment by clerk-cookie** [comment]
This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 10 days.

**Comment by ninjawarrior1337** [comment]
I don't know when this got fixed but this appears to be fixed as of clerk/javascript@4.23.2. Thanks!

**Comment by clerk-cookie** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
